### PR TITLE
Surface unavailable deployment status on providerrevision

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Updates the providerrevision post hook to return error when the
deployment is not available. This fixes the issue of a revision
reporting healthy when the deployed controller is in an unhealthy state.

Signed-off-by: hasheddan <georgedanielmangum@gmail.com>

Partially addresses issues described in #1875 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

1. Installed Crossplane
2. Installed `provider-failure`

```
k crossplane install provider hasheddan/provider-failure:v0.0.1
```

3. Observe unhealthy `Provider` / `ProviderRevision` and corresponding events

```
29s         Warning   SyncPackage               providerrevision/provider-failure-131a4ad491b9   cannot run post establish hook for package: provider package deployment is unavailable: Deployment does not have minimum availability.
```
```
NAME                            HEALTHY   REVISION   IMAGE                               STATE    AGE
provider-failure-131a4ad491b9   False     1          hasheddan/provider-failure:v0.0.1   Active   47s
```

```
NAME               INSTALLED   HEALTHY   PACKAGE                             AGE
provider-failure   True        False     hasheddan/provider-failure:v0.0.1   3m20s
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by hasheddan** [comment]
Hmm this looks like a golangci-lint bug :thinking: `level=error msg="Running error: buildir: analysis skipped: errors in package: [/home/runner/work/crossplane/crossplane/pkg/xpkg/lint_test.go:262:10: undeclared name: crd]"`

File: pkg/controller/pkg/revision/hook_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/pkg/errors"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (negz) [inline pkg/controller/pkg/revision/hook_test.go:27]: This looks like a dupe of the above `corev1`.

File: pkg/controller/pkg/revision/hook.go [hunk 2]
```diff
@@ -105,6 +108,15 @@ func (h *ProviderHooks) Post(ctx context.Context, pkg runtime.Object, pr v1alpha
 		return errors.Wrap(err, errApplyProviderDeployment)
 	}
 	pr.SetControllerReference(runtimev1alpha1.Reference{Name: d.GetName()})
+
+	for _, c := range d.Status.Conditions {
+		if c.Type == appsv1.DeploymentAvailable {
+			if c.Status == v1.ConditionTrue {
+				return nil
+			}
```
Comment (negz) [inline pkg/controller/pkg/revision/hook.go:116]: Do we have an existing helper for this - `resource.IsConditionTrue` iirc?

**Review by negz — APPROVED** [review]

File: pkg/controller/pkg/revision/hook_test.go (hunk 1 above)
Comment (hasheddan) [reply pkg/controller/pkg/revision/hook_test.go]: ah good catch! fixed :+1:

File: pkg/controller/pkg/revision/hook.go (hunk 2 above)
Comment (hasheddan) [reply pkg/controller/pkg/revision/hook.go:116]: Unfortunately our helper only works for our `controller-runtime` conditions

**Pull request merged**